api/src/main/java/org/pmiops/workbench/db/dao/UserServiceImpl.java

```diff
@@ -211,30 +213,82 @@ private void updateUserAccessTiers(DbUser dbUser, Agent agent) {
     tiersForRemoval.forEach(tier -> accessTierService.removeUserFromTier(dbUser, tier));
   }
 
-  public boolean isCompleteAndNotExpired(Timestamp completionTime) {
-    Timestamp expirationTime =
-        new Timestamp(
-            clock.millis()
-                - TimeUnit.MILLISECONDS.convert(
-                    configProvider.get().accessRenewal.expiryDays, TimeUnit.DAYS));
-    if (configProvider.get().access.enableAccessRenewal) {
-      return completionTime != null && expirationTime.before(completionTime);
+  private class ModuleTimes {
+    public Optional<Timestamp> completion;
+    public Optional<Timestamp> bypass;
+
+    public ModuleTimes(Timestamp nullableCompletion, Timestamp nullableBypass) {
+      completion = Optional.ofNullable(nullableCompletion);
+      bypass = Optional.ofNullable(nullableBypass);
+    }
+
+    public boolean isBypassed() {
+      return bypass.isPresent();
+    }
+
+    public boolean isComplete() {
+      return completion.isPresent();
     }
-    return completionTime != null;
+
+    public Optional<Timestamp> getExpiration() {
+      if (isBypassed() || !configProvider.get().access.enableAccessRenewal) {
+        return Optional.empty();
+      }
+      Long expiryDays = configProvider.get().accessRenewal.expiryDays;
+      if (expiryDays == null) {
+        throw new RuntimeException("config value accessRenewal.expiryDays.expiryDays is null");
```

Comment:
Also for follow-up: I disagree with this.

Reply:
Given this is a compliance requirement I agree this should throw if it is not configured correctly. I have had some issues with codacy as well. It is another tool that seems to get in the way a bit.

Reply:
I think Codacy's complaint here is about the specific exception - `RuntimeException` is pretty generic.  We generally prefer to use exceptions corresponding to the HTML error codes.  For this one I think 500 makes sense, which I think is `ServerErrorException`

Reply:
Yep.. agree with the warning in this case. Throwing `RuntimeException` is very rarely the correct thing to do. In this case, `ServerException` would be fine. `Preconditions.checkNotNull(..)` is also a less commonly used pattern in the codebase, but might be clearer. 

@dmohs on Codacy in general, please do not ignore these presubmit errors. This is the static analysis tool that was chosen to meet a compliance requirement. The resolution should either be to address the issue, or start a thread in slack to change the style settings. More of these may pop up in the near term, and I've asked the team to please help reconfigure Codacy to make it work for our codebase. https://pmi-engteam.slack.com/archives/CHRN2R51N/p1620758246168800

Reply:
@calbach `Preconditions.checkNotNull` matches the intent better, thank you. https://github.com/all-of-us/workbench/pull/5011

Note that I did not ignore the presubmit error, but made a judgment to forgo the requested change and documented this judgment. With respect to compliance, this is almost always accepted behavior.